src/wallet/walletutil.cpp

```diff
@@ -57,10 +57,12 @@ bool IsBerkeleyBtree(const fs::path& path)
 std::vector<fs::path> ListWalletDir()
 {
     const fs::path wallet_dir = GetWalletDir();
-    const size_t offset = wallet_dir.string().size() + 1;
     std::vector<fs::path> paths;
     boost::system::error_code ec;
 
+    // account for possible trailing backslash, since we might have only boost 1.47 we could not use fs::relative
```

Comment:
Now we have boost 1.58 (#19667).

Reply:
Not in supported stable versions.

Reply:
I guessed it might be backported at least to 0.19 and in 0.18 the bug is only int qt gui so 1.47 might be ok 
 
But In hindsight and since u look at this, the comment at line 73  is not changed be the PR and looks to me like a lingering Déjà vu with the GCC bug? 
Should we change or remove that, so that later no one grabs that trap when we change to std fs::?
line 73:
// This can be replaced by **boost::filesystem::lexically_relative** once boost is bumped to 1.60.

Reply:
I think backporting pulls should consider boost 1.47, not this one.

Reply:
It might be better if the comment said what version is requried, as opposed to the version that we have right now

Reply:
Wait, how does this only affect GUI??

Reply:
```suggestion
    // account for possible trailing backslash,  if we would have bumped to boost 1.60 we could use a simple  fs::relative call
```